wellregistry/registry/admin/monitoring_location.py

```diff
@@ -216,11 +217,95 @@ def _has_permission(perm, user, obj=None):
 
     return user.has_perm(perm) and (not obj or obj.agency.agency_cd in _get_groups(user))
 
+def _get_usergroups(request):
+    user_groups = ""
+    if not request.user.is_superuser:
+        user_groups = re.search("\[(.*)\]", str(_get_groups(request.user))).group(1)
+        user_groups = " where agency.agency_cd IN (" + user_groups + ") "
+    return user_groups
+
+def _get_where_clause(request, user_groups):
+    where_clause = ""
+    sep = ""
+    query_params = request.body.decode().split("&")
+    for query_param in query_params:
+        param = query_param.split("=")
+        if param[0] == "_selected_action":
```

Comment:
Why is this if clause necessary? This is only called from download_monitoring_locations.

Reply:
because there are multiple actions to select for, when reviewing how the body looked, each time, I determined to make the split at _selected_action to get the where clause

it returns just the selected rows, and made it easy to parse this way

all in all, the performance of download is now greatly improved, 20 to 30 seconds+
to under 1 second for the records I tried to download

Reply:
So you are getting the ids of the monitoring locations selected. Perhaps some clearer variable names would make this easier to understand.

Reply:
fixed variable names